scripts/e2e-tests.sh

```diff
@@ -143,6 +143,21 @@ function dump_cluster_state() {
   echo "***************************************"
 }
 
+# On a Prow job, save some metadata about the test for Testgrid.
+function save_metadata() {
+  (( ! IS_PROW )) && return
+  cat << EOF > ${ARTIFACTS}/metadata.json
+{
+  "Region": "${E2E_CLUSTER_REGION}",
+  "Zone": "${E2E_CLUSTER_ZONE}",
+  "Machine": "${E2E_CLUSTER_MACHINE}",
+  "Version": "${E2E_CLUSTER_VERSION}",
```

Comment:
This will be "latest" most of the time. The real version number is computed by `download_k8s()`. We should repurpose this variable to contain the real value, and create `E2E_CLUSTER_DEFAULT_VERSION` to hold the default test cluster version (it's defined in library.sh).